browser/typescript/v2/react-app/src/ampli/index.ts

```diff
@@ -0,0 +1,839 @@
+      return false;
+    }
+    return !this.disabled;
+  }
+
+  /**
+   * Initialize the Ampli SDK. Call once when your application starts.
+   * @param options Configuration options to initialize the Ampli SDK with.
+   */
+  load(options?: LoadOptions) {
+    this.disabled = options?.disabled ?? false;
+
+    if (this.isLoaded) {
+      console.warn('WARNING: Ampli is already initialized. Ampli.load() should be called once at application startup.');
+      return;
+    }
+
+    const env = options?.environment ?? 'development';
+    const apiKey = options?.client?.apiKey ?? ApiKey[env];
+
+    if (options?.client?.instance) {
+      this.amplitude = options?.client?.instance;
+    } else if (apiKey) {
+      this.amplitude = amplitude;
+      return this.amplitude?.init(apiKey, undefined, { ...DefaultOptions, ...options?.client?.options });
+    } else {
+      console.error("ERROR: ampli.load() requires 'environment', 'client.apiKey', or 'client.instance'");
+    }
+  }
+
+  /**
+   * Identify a user and set user properties.
+   *
+   * @param userId The user's id.
+   * @param properties The user properties.
+   * @param options Optional event options.
+   */
+  identify(
+    userId: string | undefined,
+    properties: IdentifyProperties,
+    options?: IdentifyOptions,
+  ) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    const event: IdentifyEvent = {
+      event_type: SpecialEventType.IDENTIFY,
+      user_properties: properties,
+      user_id: userId || options?.user_id,
+      device_id: options?.device_id
+    };
+
+    if (event.user_id) {
+      this.amplitude?.setUserId(event.user_id);
+    }
+    if (event.device_id) {
+      this.amplitude?.setDeviceId(event.device_id);
+    }
+
+    const amplitudeIdentify = new amplitude.Identify();
+    if (properties != null) {
+      for (const [key, value] of Object.entries(properties)) {
+        amplitudeIdentify.set(key, value);
+      }
+    }
+    return this.amplitude?.identify(
+      amplitudeIdentify,
+      options,
+    );
+  }
+
+  setGroup(name: string, value: string | string[]) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    this.amplitude?.setGroup(name, value);
+  }
+
+  /**
+   * Identify a group and set group properties.
+   *
+   * @param groupType The group type.
+   * @param groupName The group name.
+   * @param properties The group properties.
+   * @param options Optional event options.
+   */
+  groupIdentify(
+    groupType: string,
+    groupName: string | string[],
+    properties: GroupProperties,
+    options?: GroupOptions,
+  ) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    const event: GroupEvent = {
+      event_type: SpecialEventType.GROUP_IDENTIFY,
+      group_properties: properties,
+      user_id: options?.user_id,
+      device_id: options?.device_id
+    };
+
+    if (event.user_id) {
```

Comment:
Should we move this in core SDK? cc: @justin-fiedler

Reply:
Currently TS SDK identify method does not save user id to browser client. I think we need to add it into Core SDK.

Reply:
Personally I think we should move it to the core. I like what @yuhao900914 did in Kotlin where core identify method checks EvenOptions userId and deviceId and update them as necessary. 

Current Amplitude SDKs don't support this (user need to call setUserId directly in addition to identify) but now that we have EventOptions it seems convenient to allow users to set both at once.

Curious if others agree
cc @kevinpagtakhan @qingzhuozhen @bohan-amplitude

Reply:
Agree. I removed this part from Ampli and we should align with Kotlin https://github.com/amplitude/Amplitude-Kotlin/commit/2c8b3f1160f8503d827ca7367ada9da57836e968